apps/cowswap-frontend/src/common/pure/PendingOrderNotification/index.tsx

```diff
@@ -28,26 +28,60 @@ const ORDER_TYPE_TITLES: Record<UiOrderType, string> = {
   [UiOrderType.TWAP]: 'TWAP order',
 }
 
-interface PendingOrderNotificationProps {
+export interface PendingOrderNotificationProps {
+  account: string
+  chainId: SupportedChainId
   orderId: string
   kind: OrderKind
   orderType: UiOrderType
   inputAmount: CurrencyAmount<Token>
   outputAmount: CurrencyAmount<Token>
   receiver?: string
   orderCreationHash?: string
+  isSafeWallet: boolean
 }
 
-export function PendingOrderNotification({
-  orderId,
-  kind,
-  orderType,
-  inputAmount,
-  outputAmount,
-  receiver,
-}: PendingOrderNotificationProps) {
-  const { chainId, account } = useWalletInfo()
-  const isSafeWallet = useIsSafeWallet()
+export function getPendingOrderNotificationToast(props: PendingOrderNotificationProps): OnToastMessagePayload | null {
+  const { orderId, kind, inputAmount, outputAmount, receiver, account, orderType } = props
+
+  if (!account) return null
+
+  const toAddress = receiver && isAddress(receiver) ? shortenAddress(receiver) : receiver
+
+  const inputAmountElement = formatTokenAmountWithSymbol({
+    amount: inputAmount,
+    tokenSymbol: inputAmount.currency,
+  })
+  const outputAmountElement = formatTokenAmountWithSymbol({
+    amount: outputAmount,
+    tokenSymbol: outputAmount.currency,
+  })
+
+  const messagePrefix = `${ORDER_TYPE_TITLES[orderType]} submitted: `
+
+  const baseMessage = (() => {
+    const isSellOrder = kind === OrderKind.SELL
+
+    if (isSellOrder) {
+      return `Sell ${inputAmountElement} for at least ${outputAmountElement}`
+    } else {
+      return `Buy ${outputAmountElement} for at most ${inputAmountElement}`
+    }
+  })()
```

Comment:
Nitpick: do a helper fn instead an inline one

Reply:
I'm happy to do if you prefer, but I did it IIFE because the function is simple enough, and this way the logic on how we compose the message is closer together. Its more analogous to the code in the component version of it

I know you don't love this pattern, just confirm me you feel I should do it, and I will move it. Otherwise I slightly prefer to have it self contained and specially this part is the gist of the function, so i prefer to not move it away